examples/sampling/pyg/test.py

```diff
@@ -0,0 +1,167 @@
+This flowchart describes the main functional sequence of the provided example.
+main: 
+
+main
+│
+├───> Load and preprocess dataset (GraphBolt)
+│     │
+│     └───> Utilize GraphBolt's BuiltinDataset for dataset handling
+│
+├───> Instantiate the SAGE model (PyTorch Geometric)
+│     │
+│     └───> Define the GraphSAGE model architecture
+│
+├───> Train the model
+│     │
+│     ├───> Mini-Batch Processing with GraphBolt
+│     │     │
+│     │     └───> Efficient handling of mini-batches using GraphBolt's utilities
+│     │
+│     └───> Training Loop
+│           │
+│           ├───> Forward and backward passes
+│           │
+│           └───> Parameters optimization
+│
+└───> Evaluate the model
+      │
+      └───> Performance assessment on validation and test datasets
+            │
+            └───> Accuracy and other relevant metrics calculation
+
+
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.graphbolt import (
+    BuiltinDataset,
+    CopyTo,
+    DataLoader,
+    FeatureFetcher,
+    ItemSampler,
+)
+from torch_geometric.nn import GCNConv
+
+
+dataset = BuiltinDataset("ogbn-arxiv").load()
+graph = dataset.graph
+feature = dataset.feature
+train_set = dataset.tasks[0].train_set
+valid_set = dataset.tasks[0].validation_set
+test_set = dataset.tasks[0].test_set
+num_classes = dataset.tasks[0].metadata["num_classes"]
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def create_sampler_and_loader(dataset_set):
+    sampler = ItemSampler(dataset_set, batch_size=1024, shuffle=True)
+    datapipe = sampler.sample_neighbor(graph, [4, 4])
+    datapipe = FeatureFetcher(datapipe, feature, node_feature_keys=["feat"])
+    datapipe = CopyTo(datapipe, device=device)
+    return DataLoader(datapipe, num_workers=0)
+
+
+train_dataloader = create_sampler_and_loader(train_set)
+valid_dataloader = create_sampler_and_loader(valid_set)
+test_dataloader = create_sampler_and_loader(test_set)
+
+
+class GCN(torch.nn.Module):
+    def __init__(self, in_channels, hidden_channels, out_channels):
+        super(GCN, self).__init__()
+        self.conv1 = GCNConv(in_channels, hidden_channels)
+        self.conv2 = GCNConv(hidden_channels, out_channels)
+
+    def forward(self, x, edge_index):
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = F.dropout(x, training=self.training)
+        x = self.conv2(x, edge_index)
+        return F.log_softmax(x, dim=1)
+
+
+in_channels = feature.size("node", None, "feat")[0]
+model = GCN(in_channels, 16, num_classes)
+model = model.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
+
+
+def train_epoch(model, dataloader):
+    model.train()
+    total_loss = 0
+    total_correct = 0
+    total_samples = 0
+    batch_count = 0
+
+    for minibatch in dataloader:
+        batch_count += 1
+        node_features = minibatch.node_features["feat"].to(device)
+        block = minibatch.blocks[-1]
```

Comment:
only the last block is used for model forward?

Reply:
just fixed this